### PR TITLE
feature: collage.json generator

### PR DESCRIPTION
**Pull request by fmiccolis**
#### Prerequisites checklist

- [ ] I have read the [contributing guidelines](https://github.com/PhotoboothProject/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Other, please explain:

#### What changes did you make? (Give an overview)
I've created a wysiwyg page to visually generate the collage.json file.
I've also extended the functions of the file by adding:
- background color
- placeholder image
- single image frame toggle

The image upload route is now a file upload because i give the user the possibility to upload .ttf fonts inside /private/fonts folder. The dispatch from /private/fonts and /private/images is done by checking the file type before appending the folderName.

#### Is there anything you'd like reviewers to focus on?
I need an hand from the community to test edge cases

### CODE REVIEW

File: src/Collage.php [hunk 1]
```diff
@@ -809,14 +793,6 @@ public static function createCollage(array $config, array $srcImagePaths, string
             }
         }
 
-        // Rotate image if needed
-        if ($rotate_after_creation) {
```
Comment (andi34) [inline src/Collage.php]: We still need this for the other layouts

**Review by andi34 — CHANGES_REQUESTED** [review]

**Review by andi34 — APPROVED** [review]

**Comment by andi34** [comment]
While saving the layout via generator the collage config should be updated too. ATM e.g. the limits are not adjusted based on the given images array and we would need to go back to the Adminpanel first and save the config again before the layout can be tested.

**Comment by benjaminkott** [comment]
Hey @andi34, hey @fmiccolis a quick update.

I´ve extracted some of the changes to make the pull-request more manageable. 
This pr should now only include the collage generation.

feature: add font selector
https://github.com/PhotoboothProject/photobooth/commit/b771215d3eea7de2d95eab4128c536b43790bc03

feature: allow font uploads
https://github.com/PhotoboothProject/photobooth/commit/5ffca06db4709aa243ca916103e395d24198acc4

I´ve merged the changes into this branch, as a rebase was too complicated.
This branch is now up2date with current upstream/dev.

**Comment by andi34** [comment]
https://github.com/PhotoboothProject/photobooth/pull/709

Was reverted by accident
